Fails to build on Fedora 43

### Thanks in advance for your bug report!

- [x] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [x] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [x] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [x] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [x] Have you checked you are on the latest release of Pulsar?

### What happened?

When running `yarn install` in the Pulsar root directory on Fedora 43, I got this error message:

```
make: Entering directory '/home/andrew/Code/pulsar/node_modules/@pulsar-edit/fuzzy-native/build'
  CXX(target) Release/obj.target/fuzzy-native/src/binding.o
../src/binding.cpp: In static member function ‘static void Matcher::Match(const Nan::FunctionCallbackInfo<v8::Value>&)’:
../src/binding.cpp:155:21: warning: ignoring return value of ‘v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)’, declared with attribute ‘warn_unused_result’ [-Wunused-result]
  155 |           array->Set(context, i, New(match.matchIndexes->at(i)));
      |           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/andrew/.cache/node-gyp/20.11.1/include/node/v8-array-buffer.h:13,
                 from /home/andrew/.cache/node-gyp/20.11.1/include/node/v8.h:24,
                 from /home/andrew/.cache/node-gyp/20.11.1/include/node/node.h:73,
                 from ../../../nan/nan.h:62,
                 from ../src/binding.cpp:1:
/home/andrew/.cache/node-gyp/20.11.1/include/node/v8-object.h:249:37: note: declared here
  249 |   V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
      |                                     ^~~
../src/binding.cpp:159:18: warning: ignoring return value of ‘v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)’, declared with attribute ‘warn_unused_result’ [-Wunused-result]
  159 |       result->Set(context, result_count++, obj);
      |       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/andrew/.cache/node-gyp/20.11.1/include/node/v8-object.h:249:37: note: declared here
  249 |   V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
      |                                     ^~~
  CXX(target) Release/obj.target/fuzzy-native/src/fuzzaldrin_score.o
../src/fuzzaldrin_score.cpp: In function ‘bool queryIsLastPathSegment(const std::string&, const std::string&)’:
../src/fuzzaldrin_score.cpp:59:12: warning: unused variable ‘res2’ [-Wunused-variable]
   59 |       auto res2 = string.rfind(query) == (string.length() - query.length());
      |            ^~~~
../src/fuzzaldrin_score.cpp:56:10: warning: unused variable ‘res’ [-Wunused-variable]
   56 |     auto res = string[string.length() - query.length() - 1] == PathSeparator[0];
      |          ^~~
../src/fuzzaldrin_score.cpp: In function ‘float fuzzaldrin_score(const std::string&, const std::string&)’:
../src/fuzzaldrin_score.cpp:110:10: warning: unused variable ‘div’ [-Wunused-variable]
  110 |     auto div = (queryLength / (float) stringLength);
      |          ^~~
  CXX(target) Release/obj.target/fuzzy-native/src/score_match.o
  CXX(target) Release/obj.target/fuzzy-native/src/MatcherBase.o
In file included from ../src/MatcherBase.cpp:1:
../src/MatcherBase.h:21:3: error: ‘uint32_t’ does not name a type
   21 |   uint32_t id;
      |   ^~~~~~~~
../src/MatcherBase.h:7:1: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
    6 | #include <vector>
  +++ |+#include <cstdint>
    7 | 
../src/MatcherBase.h:31:15: error: ‘uint32_t’ has not been declared
   31 |               uint32_t id,
      |               ^~~~~~~~
../src/MatcherBase.h:31:15: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
../src/MatcherBase.h: In constructor ‘MatchResult::MatchResult(float, int, int, const std::string*)’:
../src/MatcherBase.h:34:7: error: class ‘MatchResult’ does not have any field named ‘id’
   34 |       id(id),
      |       ^~
../src/MatcherBase.h: At global scope:
../src/MatcherBase.h:55:5: error: ‘uint32_t’ does not name a type
   55 |     uint32_t id;
      |     ^~~~~~~~
../src/MatcherBase.h:55:5: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
../src/MatcherBase.h:68:5: error: ‘uint64_t’ does not name a type
   68 |     uint64_t bitmask;
      |     ^~~~~~~~
../src/MatcherBase.h:68:5: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
../src/MatcherBase.h:80:21: error: ‘uint32_t’ has not been declared
   80 |   void addCandidate(uint32_t id, const std::string &candidate);
      |                     ^~~~~~~~
../src/MatcherBase.h:80:21: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
../src/MatcherBase.h:81:24: error: ‘uint32_t’ has not been declared
   81 |   void removeCandidate(uint32_t id);
      |                        ^~~~~~~~
../src/MatcherBase.h:81:24: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
../src/MatcherBase.h:91:22: error: ‘uint32_t’ was not declared in this scope
   91 |   std::unordered_map<uint32_t, size_t> lookup_;
      |                      ^~~~~~~~
../src/MatcherBase.h:91:22: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
../src/MatcherBase.h:91:38: error: template argument 1 is invalid
   91 |   std::unordered_map<uint32_t, size_t> lookup_;
      |                                      ^
../src/MatcherBase.h:91:38: error: template argument 3 is invalid
../src/MatcherBase.h:91:38: error: template argument 4 is invalid
../src/MatcherBase.h:91:38: error: template argument 5 is invalid
../src/MatcherBase.cpp: In function ‘void thread_worker(const std::string&, const std::string&, const MatchOptions&, bool, std::atomic<float>*, size_t, std::vector<MatcherBase::CandidateData>&, size_t, size_t, ResultHeap&)’:
../src/MatcherBase.cpp:150:30: error: ‘struct MatcherBase::CandidateData’ has no member named ‘bitmask’
  150 |     if ((bitmask & candidate.bitmask) == bitmask) {
      |                              ^~~~~~~
../src/MatcherBase.cpp:172:21: error: ‘struct MatcherBase::CandidateData’ has no member named ‘id’
  172 |           candidate.id,
      |                     ^~
../src/MatcherBase.cpp: In member function ‘std::vector<MatchResult> MatcherBase::findMatches(const std::string&, const MatcherOptions&)’:
../src/MatcherBase.cpp:271:15: error: ‘const struct MatchResult’ has no member named ‘id’
  271 |           top.id,
      |               ^~
../src/MatcherBase.cpp: At global scope:
../src/MatcherBase.cpp:289:6: error: no declaration matches ‘void MatcherBase::addCandidate(uint32_t, const std::string&)’
  289 | void MatcherBase::addCandidate(uint32_t id, const string &candidate) {
      |      ^~~~~~~~~~~
../src/MatcherBase.h:80:8: note: candidate is: ‘void MatcherBase::addCandidate(int, const std::string&)’
   80 |   void addCandidate(uint32_t id, const std::string &candidate);
      |        ^~~~~~~~~~~~
../src/MatcherBase.h:52:7: note: ‘class MatcherBase’ defined here
   52 | class MatcherBase {
      |       ^~~~~~~~~~~
../src/MatcherBase.cpp:305:6: error: no declaration matches ‘void MatcherBase::removeCandidate(uint32_t)’
  305 | void MatcherBase::removeCandidate(uint32_t id) {
      |      ^~~~~~~~~~~
../src/MatcherBase.h:81:8: note: candidate is: ‘void MatcherBase::removeCandidate(int)’
   81 |   void removeCandidate(uint32_t id);
      |        ^~~~~~~~~~~~~~~
../src/MatcherBase.h:52:7: note: ‘class MatcherBase’ defined here
   52 | class MatcherBase {
      |       ^~~~~~~~~~~
../src/MatcherBase.cpp: In member function ‘void MatcherBase::clear()’:
../src/MatcherBase.cpp:319:11: error: request for member ‘clear’ in ‘((MatcherBase*)this)->MatcherBase::lookup_’, which is of non-class type ‘int’
  319 |   lookup_.clear();
      |           ^~~~~
../src/MatcherBase.cpp: In member function ‘void MatcherBase::reserve(size_t)’:
../src/MatcherBase.cpp:324:11: error: request for member ‘reserve’ in ‘((MatcherBase*)this)->MatcherBase::lookup_’, which is of non-class type ‘int’
  324 |   lookup_.reserve(n);
      |           ^~~~~~~
make: *** [fuzzy-native.target.mk:113: Release/obj.target/fuzzy-native/src/MatcherBase.o] Error 1
make: Leaving directory '/home/andrew/Code/pulsar/node_modules/@pulsar-edit/fuzzy-native/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/home/andrew/.asdf/installs/nodejs/20.11.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
gyp ERR! System Linux 6.17.1-300.fc43.aarch64
gyp ERR! command "/home/andrew/.asdf/installs/nodejs/20.11.1/bin/node" "/home/andrew/.asdf/installs/nodejs/20.11.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/andrew/Code/pulsar/node_modules/@pulsar-edit/fuzzy-native
```

I don't get this on macOS.

Claude tells me that this is because GCC is stricter than Clang about implicit includes. Running

```sh
export CXXFLAGS="-include cstdint"; yarn install
```

seems to fix this, but the better solution is apparently to explicitly add

```
#include <cstdint>
```

in `MatcherBase.h`.

I might make this change later myself, but I'm opening this ticket just in case I forget.
```

### Pulsar version

1.131.0 (master branch)

### Which OS does this happen on?

🐧 Red Hat based (Fedora, Alma, RockyLinux, CentOS Stream, etc.)

### OS details

Fedora 43

### Which CPU architecture are you running this on?

Apple M-series

### What steps are needed to reproduce this?

* Check out the source code and install needed dependencies, [as described here](https://github.com/pulsar-edit/pulsar/issues/1394#issuecomment-3700473182)
* Run `yarn install`

### Additional Information:

_No response_

--- a/src/MatcherBase.h
+++ b/src/MatcherBase.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <memory>
 #include <string>
 #include <unordered_map>

